packages/perseus-editor/src/article-editor.tsx

```diff
@@ -64,11 +70,24 @@ export default class ArticleEditor extends React.Component<Props, State> {
 
     constructor(props: Props) {
         super(props);
+
+        // Check if the json needs to be converted
+        const conversionWarningRequired = conversionRequired(
+            props.json as PerseusRenderer,
+        );
+        let json = props.json;
+
+        // Convert the json if needed
+        if (conversionWarningRequired) {
+            json = Array.isArray(props.json)
```

Comment:
If you use `props.json instanceof Array`. That will still do array checking but properly narrow `props.json` so you don't need the `as`.